packages/auth/src/grant/routes.ts

```diff
@@ -102,6 +108,27 @@ async function createGrantInitiation(
     return
   }
 
+  if (
+    !deps.config.incomingPaymentInteraction &&
+    body.access_token.access
+      .map((acc) => {
+        return isIncomingPaymentAccessRequest(acc as IncomingPaymentRequest)
```

Comment:
I didn't want to touch this because I thought I would add the change to have the `ctx.body` typed but that's probably going to be a subsequent PR so yes, I'll change that.

Reply:
`BaseAccessRequest` might work, but I just noticed the other type guards also have the target type as the argument type.
Maybe all they can all be updated as part of #351 instead of this pr.

Reply:
I think that is a good idea.